src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex.ExtensionMethodInfo.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FindSymbols
+{
+    internal partial class SyntaxTreeIndex
+    {
+        private readonly struct ExtensionMethodInfo
+        {
+            // We devide extension methods into two categories, simple and complex, for filtering purpose.
+            // Whether a method is simple is determined based on if we can determine it's target type easily
+            // with a pure text matching. For complex methods, we will need to rely on symbol to decide if it's 
+            // feasible.
+            //
+            // Complex methods include:
+            // - Method declared in the document which includes using alias directive
```

Comment:
note: this would not be too hard to support.

Reply:
Fixed now

Reply:
nice!  can we update the commen then? :)